libdnf/hy-util.cpp

```diff
@@ -127,10 +127,12 @@ hy_detect_arch(char **arch)
         char *modifier = un.machine + 5;
         while(isdigit(*modifier)) /* keep armv7, armv8, armv9, armv10, armv100, ... */
             modifier++;
+#if !defined(__FreeBSD__)
         if (getauxval(AT_HWCAP) & HWCAP_ARM_VFP)
             *modifier++ = 'h';
         if ((atoi(un.machine+4) == 7) && (getauxval(AT_HWCAP) & HWCAP_ARM_NEON))
             *modifier++ = 'n';
+#endif
```

Comment:
Why isn't this applicable on FreeBSD?
Is it because `getauxval()` or `AT_HWCAP` is not available?
Or is it because the 'h' and 'n' ARM extensions are not supported?

Reply:
This is actually a problem, though, because that means we can't map architectures correctly based on CPU capabilities.

Reply:
@Conan-Kudo Any suggestion?
Anyway we have to make a decision what to do with PR.

 @yurivict There is still not answered question - `Why isn't this applicable on FreeBSD?
Is it because getauxval() or AT_HWCAP is not available?` Please can you provide additional information?

Reply:
I don't know what the equivalents of these APIs would be for FreeBSD, so I don't know. But we can't _not_ have that. What does RPM do here?

Reply:
In DNF5 we replaced the detection by a call of `rpmGetArchInfo(&value, nullptr)` from `rpmlib.h`